pkg/search/controller.go

```diff
@@ -331,7 +331,22 @@ func (c *Controller) updateResourceRegistry(oldObj, newObj interface{}) {
 
 // deleteResourceRegistry parse the resourceRegistry object and add deleted Cluster to the queue
 func (c *Controller) deleteResourceRegistry(obj interface{}) {
-	rr := obj.(*searchv1alpha1.ResourceRegistry)
+	rr, isRR := obj.(*searchv1alpha1.ResourceRegistry)
+	// We can get DeletedFinalStateUnknown instead of *clusterV1alpha1.Cluster here and
```

Comment:
clusterV1alpha1.Cluster -> searchv1alpha1.ResourceRegistry

Reply:
@XiShanYongYe-Chang The code comment has been fixed.

I think this patch need to cherry-pick the previous branch, and need to check the other controller `cache.ResourceEventHandlerFuncs` `DeleteFunc` dealt with the problem.

Reply:
> I think this patch need to cherry-pick the previous branch and need to check whether other controller of `DeleteFunc` handled `DeletedFinalStateUnknown` types of object.

Thanks~
Can you help do it?

Reply:
> > I think this patch need to cherry-pick the previous branch and need to check whether other controller of `DeleteFunc` handled `DeletedFinalStateUnknown` types of object.
> 
> Thanks~ Can you help do it?

Okay, I will deal with this issue later.